pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -598,6 +601,10 @@ public void execute(Query[] queries, @Nullable ParameterList[] parameterLists,
     }
   }
 
+  private boolean possibleIsolationLevelQuery(Query query) {
+    return query.getNativeSql().toLowerCase(Locale.ROOT).contains("isolation");
```

Comment:
This would result in parsing overhead for every query execution. I would suggest moving this to something like `org.postgresql.core.CachedQueryCreateAction#create` or `org.postgresql.core.Parser#parseJdbcSql` so we don't end up with repeated `toLowerCase().contains()` calls